data_structures/doubly_linked_list.ts

```diff
@@ -0,0 +1,299 @@
+/**
+ * This is an implementation of a Doubly Linked List.
+ * A Doubly Linked List is a data structure that contains a head, tail and length property.
+ * Linked Lists consist of nodes, and each node has a value and a pointer to the next and previous node (can be null).
+ * The time complexity of the operations is O(n).
```

Comment:
This is not true. The time complexity of many operations (insert / remove at front / back) is O(1), which is the entire raison d'être of linked lists.